File: src/main/kotlin/studio/pinkcloud/voyager/deployment/cloudflare/responses/AddDnsRecord.kt
package studio.pinkcloud.voyager.deployment.cloudflare.responses

import kotlinx.serialization.Serializable


// https://developers.cloudflare.com/api/operations/dns-records-for-a-zone-create-dns-record

@Serializable
data class Meta(
    val auto_added: Boolean,
    val source: String
)

@Serializable
data class AddDnsRecordData(
    val content: String, // ipv4 address
    val name: String,
    val proxied: Boolean,
    val type: String,
    val comment: String,
    val id: String,
    val locked: Boolean,
    val meta: Meta,
    val modified_on: String,
    val proxiable: Boolean,
    val tags: Array<String>,
    val ttl: Long,
    val zone_id: String,
    val zone_name: String,
)

@Serializable
data class AddDnsRecordSuccess(
    val errors: List<CloudflareError>,
    val messages: List<CloudflareMessage>,
    val result: AddDnsRecordData
)

@Serializable
data class AddDnsRecordFailure(
    val errors: List<CloudflareError>,
    val messages: List<CloudflareMessage>,
)

File: src/modules/cloudflare/types/add_dns_record.rs
use serde::Deserialize;
use crate::modules::cloudflare::types::cloudflare_responses::CloudflareError;

#[derive(Debug, Deserialize)]
pub struct OkResponse {
  // pub errors: Vec<CloudflareError>,
  // pub messages: Vec<CloudflareMessage>,
  pub result: Option<Data>,
  // pub success: bool,
}

#[derive(Debug, Deserialize)]
pub struct Data {
  pub id: String,
  // pub zone_id: String,
  // pub zone_name: String,
  // pub name: String,
  // #[serde(rename = "type")]
  // pub r#type: String,
  // pub content: String,
  // pub proxiable: bool,
  // pub proxied: bool,
  // pub ttl: i64,
  // pub locked: bool,
  // pub meta: Meta,
  // pub comment: String,
  // pub tags: Vec<String>,
  // pub created_on: String,
  // pub modified_on: String,
}
#[derive(Debug, Deserialize)]
pub struct Meta {
  // pub auto_added: bool,
  // pub managed_by_apps: bool,
  // pub managed_by_argo_tunnel: bool,
}

#[derive(Debug, Deserialize)]
pub struct FailureResponse {
  // pub messages: Vec<CloudflareMessage>,
  pub errors: Vec<CloudflareError>,
}
